src/main/java/edu/usc/ir/geo/gazetteer/GeoNameResolver.java

```diff
@@ -155,6 +178,27 @@ public GeoNameResolver(String indexPath) throws IOException {
 
 	}
 
+	public List<Location> searchNearby(Double latitude, Double longitude, Double distance, String indexerPath, int count) throws IOException {
+
+		SpatialArgs spatialArgs = new SpatialArgs(SpatialOperation.IsWithin,
+				ctx.makeCircle(longitude,latitude, distance));
+		
+		String key = latitude+"-"+longitude;
+		Filter filter = strategy.makeFilter(spatialArgs);
+		
+		IndexSearcher searcher = new IndexSearcher(createIndexReader(indexerPath));
+		Sort sort = new Sort(populationSort);
+		TopDocs topDocs = searcher.search(new MatchAllDocsQuery(), filter, hitsPerPage, sort);
+
+		ScoreDoc[] scoreDocs = topDocs.scoreDocs;
+		HashMap<String, List<Location>> allCandidates = new HashMap<String, List<Location>>();
+
+		getMatchingCandidates(searcher, allCandidates, key, scoreDocs);
+		List<Location> results = allCandidates.get(key);
+		
+		return results.subList(0, count > results.size() ? results.size() : count);
```

Comment:
If you only want as much as "count" docs back, then tell Lucene that in the searcher.search() method.